ui/documents/semantic-ui/js/documents/forms/components/SubjectsField/SubjectsField.jsx

```diff
@@ -1,50 +1,150 @@
-import React from "react";
+import React, { useMemo, useCallback } from "react";
 import PropTypes from "prop-types";
-import { Form } from "semantic-ui-react";
-import { ArrayField } from "react-invenio-forms";
+import { Form, Icon, Divider, Button } from "semantic-ui-react";
 import { i18next } from "@translations/i18next";
-import { MultilingualTextInput, ArrayFieldItem } from "@js/oarepo_ui";
+import { SubjectsModal } from "./SubjectsModal";
+import { useFormikContext, getIn } from "formik";
+import _difference from "lodash/difference";
+import { FieldLabel } from "react-invenio-forms";
 
-export const SubjectsField = ({ fieldPath, helpText, defaultNewValue }) => {
+export const SubjectsField = ({ fieldPath }) => {
+  const { values, setFieldValue } = useFormikContext();
+  const subjects = getIn(values, fieldPath, []);
+  const externalSubjects = useMemo(
+    () => subjects.filter((subject) => subject?.subjectScheme !== "keyword"),
+    [subjects]
+  );
+
+  const regularSubjects = useMemo(
+    () =>
+      _difference(subjects, externalSubjects).map((subject) => ({
+        ...subject,
+        id: crypto.randomUUID(),
+      })),
+    [subjects, externalSubjects]
+  );
+  const handleSubjectRemoval = useCallback(
+    (id, lang) => {
+      const newRegularSubjects = regularSubjects.map((subject) => {
+        if (subject.id === id) {
+          subject.subject = subject.subject.filter((s) => s.lang !== lang);
+          return subject;
+        }
+        return subject;
+      });
+      setFieldValue(fieldPath, [
+        ...externalSubjects,
+        ...newRegularSubjects
+          .filter((subject) => subject?.subject?.length > 0)
+          .map((subject) => {
+            const { id, ...subjectWithoutId } = subject;
```

Comment:
@mirekys Zdar, I am using id, just so I would have a reliable way to remove a keyword, but here I am prepping them to be sent via API, and I cannot send crypto UUID to the backend, so I remove it before placing into formik state. Otherwise, I would have to specifically process the subjects field (removal of the random UUIDs) when submitting form and this would kind of ruin the conception.